frontend/src/metabase/visualizations/echarts/cartesian/model/dataset.ts

```diff
@@ -172,3 +172,60 @@ export const getSortedSeriesModels = (
 
   return orderedSeriesModels;
 };
+
+type ReplacerFn = (dataKey: string, value: RowValue) => RowValue;
+
+/**
+ * Creates a new dataset with the values replaced according to the provided replacer function.
+ *
+ * @param {Record<DataKey, RowValue>[]} dataset - The original dataset.
+ * @param {ReplacerFn} replacer - The function that will be used to replace values.
+ * @returns {Record<DataKey, RowValue>[]} A new dataset with the replaced values.
+ */
+export const replaceValues = (
+  dataset: Record<DataKey, RowValue>[],
+  replacer: ReplacerFn,
+) => {
+  return dataset.map(datum => {
```

Comment:
Nit: this should be called `row` not `datum`. A `datum` ([at least in d3 terminology according to the creator]([url](https://stackoverflow.com/a/10091516))) is an individual data value.